src/solc_0.8/Game/GameMinter.sol

```diff
@@ -21,54 +22,49 @@ contract GameMinter is WithMetaTransaction, IGameMinter {
 
     constructor(
         GameToken gameTokenContract,
-        address metaTransactionContract,
+        address trustedForwarder,
         uint256 gameMintingFee,
         uint256 gameUpdateFee,
         address feeBeneficiary,
         IERC20 sand
-    ) {
+    ) ERC2771Context(trustedForwarder) {
         _gameToken = gameTokenContract;
-        _setMetaTransactionProcessor(metaTransactionContract, METATX_SANDBOX);
         _gameMintingFee = gameMintingFee;
         _gameUpdateFee = gameUpdateFee;
         _feeBeneficiary = feeBeneficiary;
         _sand = sand;
     }
 
     /// @notice Function to create a new GAME token
-    /// @param from The address of the one creating the game (included in the gameId).
     /// @param to The address who will be assigned ownership of this game.
     /// @param creation The struct containing ids & ammounts of assets to add to this game,
     /// along with the uri to set.
     /// @param editor The address to allow to edit (can also be set later).
     /// @param subId A random id created on the backend.
     /// @return gameId The id of the new GAME token (erc721)
     function createGame(
-        address from,
         address to,
         GameToken.Update calldata creation,
         address editor,
         uint64 subId
     ) external override returns (uint256 gameId) {
-        require(msg.sender == from || _isValidMetaTx(from), "CREATE_ACCESS_DENIED");
-        _chargeSand(from, _gameMintingFee);
-        return _gameToken.createGame(from, to, creation, editor, subId);
+        address msgSender = _msgSender();
+        _chargeSand(msgSender, _gameMintingFee);
+        return _gameToken.createGame(msgSender, to, creation, editor, subId);
     }
 
     /// @notice Update an existing GAME token.This actually burns old token
     /// and mints new token with same basId & incremented version.
-    /// @param from The address whose GAME token is to be updated.
     /// @param gameId The current id of the GAME token.
     /// @param update The values to use for the update.
     /// @return newId The new gameId.
-    function updateGame(
-        address from,
-        uint256 gameId,
-        GameToken.Update memory update
-    ) external override returns (uint256 newId) {
-        _checkAuthorization(from, gameId);
-        _chargeSand(from, _gameUpdateFee);
-        return _gameToken.updateGame(from, gameId, update);
+    function updateGame(uint256 gameId, GameToken.Update memory update) external override returns (uint256 newId) {
+        address gameOwner = _gameToken.ownerOf(gameId);
+        address msgSender = _msgSender();
+        bool isEditor = _gameToken.isGameEditor(gameOwner, msgSender);
```

Comment:
isEditor should be called only if msgSender is not gameOnwer to avoid extra gas cost if it is the game owner

Reply:
indeed. Fixed.